packages/react/src/primitives/Radio/Radio.tsx

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import { Flex } from '../Flex';
+import { Input } from '../Input';
+import { useRadioGroupContext } from '../RadioField/context';
+import { Text } from '../Text';
+import { VisuallyHidden } from '../VisuallyHidden';
+import { RadioProps } from '../types';
+import { ComponentClassNames } from '../shared';
+
+export const Radio: React.FC<RadioProps> = ({
+  children,
+  className,
+  id,
+  isDisabled,
+  testId,
+  value,
+  ...rest
+}) => {
+  const {
+    currentValue,
+    defaultValue,
+    name,
+    hasError,
+    isGroupDisabled,
+    isRequired,
+    isReadOnly,
+    onChange,
+    size,
+  } = useRadioGroupContext();
+
+  const shouldDisabled =
+    isGroupDisabled || isDisabled || (isReadOnly && defaultValue !== value);
+
+  // for controlled component
+  const checked =
+    currentValue !== undefined ? value === currentValue : undefined;
+
+  // for uncontrolled component
+  const defaultChecked =
+    defaultValue !== undefined ? value === defaultValue : undefined;
+  return (
+    <Flex
+      as="label"
+      alignItems="center"
+      className={classNames(ComponentClassNames.Radio, className)}
+      justifyContent="flex-start"
+    >
+      <VisuallyHidden
+        as={Input}
```

Comment:
Hmm, but I don't know how you would do the focus styling, since you don't have that in context.

Reply:
You got the reason. I am using CSS sibling selector because I don't have a good solution on top of my head for uncontrolled component. For controlled component, it is easy to update `data-checked` attribute along with a re-rendering. But you cannot do that with an uncontrolled component. The `defaultChecked` only applies when component get mounted the first time. It is not supposed to be used dynamically. This also answer a styling comment below I think

Reply:
Yeah, I withdraw my change request here. I think this just reveals that we need a way to pass through the type of the `as` component. This doesn't need to be addressed in this PR, I will create a separate ticket in Asana to improve the `as` typing.

Reply:
What if instead of using `<VisuallyHidden as="input">` we used `<Input className={ComponentClassNames.VisuallyHidden}>` or something? So we get the type checking and the visually hidden stuff?

Reply:
Yea, we could. That's sounds great

Reply:
Yeah, I like that solution better too!